Sources/CoreMetrics/Metrics.swift

```diff
@@ -172,8 +172,34 @@ public class Gauge: Recorder {
     }
 }
 
-public enum TimeUnit {
-    case nanoseconds, milliseconds, seconds, minutes, hours, days
+public struct TimeUnit: Equatable {
+    private enum Code: Equatable {
+        case nanoseconds
+        case microseconds
+        case milliseconds
+        case seconds
+        case minutes
+        case hours
+        case days
+    }
+
+    private let code: Code
```

Comment:
code reads a bit weird, `encoding`?
```suggestion
    private let encoding: Encoding
```

?